morpheus/_lib/messages/module.cpp

```diff
@@ -379,13 +378,50 @@ PYBIND11_MODULE(messages, _module)
              py::arg("config"))
         .def("config", pybind11::overload_cast<ControlMessage&>(&ControlMessageProxy::config))
         .def("copy", &ControlMessageProxy::copy)
-        .def("get_metadata", &ControlMessageProxy::get_metadata, py::arg("key") = py::none())
+        .def(
+            "get_metadata",
+            [](ControlMessage& self, const py::object& key = py::none(), const py::object& default_value = py::none()) {
+                if (key.is_none())
+                {
+                    // Directly return all metadata
+                    return ControlMessageProxy::get_metadata(self, key, default_value);
+                }
+                else
+                {
+                    // Call the original get_metadata with the provided key and default value
+                    return ControlMessageProxy::get_metadata(self, key.cast<py::str>(), default_value);
+                }
+            },
+            py::arg("key")           = py::none(),
+            py::arg("default_value") = py::none())
         .def("get_tasks", &ControlMessageProxy::get_tasks)
+        .def("get_timestamp",
+             py::overload_cast<ControlMessage&, const std::string&, const std::string&>(
+                 &ControlMessageProxy::get_timestamp),
+             "Retrieve timestamps matching a regex filter within a given group.",
+             py::arg("group"),
+             py::arg("regex_filter"))
+        .def("get_timestamp",
+             py::overload_cast<ControlMessage&, const std::string&, const std::string&, bool>(
+                 &ControlMessageProxy::get_timestamp),
+             "Retrieve the timestamp for a given group and key. Returns None if the timestamp does not exist and "
+             "fail_if_nonexist is False.",
+             py::arg("group"),
+             py::arg("key"),
+             py::arg("fail_if_nonexist"))
+        .def("set_timestamp",
+             &ControlMessageProxy::set_timestamp,
+             "Set a timestamp for a given key and group.",
+             py::arg("group"),
+             py::arg("key"),
+             py::arg("timestamp_ns"))
         .def("has_metadata", &ControlMessage::has_metadata, py::arg("key"))
         .def("has_task", &ControlMessage::has_task, py::arg("task_type"))
         .def("list_metadata", &ControlMessageProxy::list_metadata)
         .def("payload", pybind11::overload_cast<>(&ControlMessage::payload), py::return_value_policy::move)
         .def("payload", pybind11::overload_cast<const std::shared_ptr<MessageMeta>&>(&ControlMessage::payload))
+        .def("tensors", pybind11::overload_cast<>(&ControlMessage::tensors), py::return_value_policy::move)
```

Comment:
The return value policy needs to reference the parent object. I believe its called `reference_internal`

Reply:
So, reference_internal ties the lifetime of the returned object to the parent. I'm not sure this is the correct behavior.

If we create the tensor_memory, then pass it to the CM, then get it back in some other context, its possible we could have 3 references to the tensor_memory, and any of them should be able to safely go out of scope.

If we create the tensor_memory SP in c++, and then retrieve it in python at a later time, we should still be fine regardless of the order in which they go out of scope.